include/fiction/algorithms/simulation/sidb/clustercomplete.hpp

```diff
@@ -0,0 +1,513 @@
+     * exhaustively. The sets of SiDBs that witness local population stability for each respective charge state may be
+     * partitioned into disjoint sets such that the number of required witnesses for each respective charge state is
+     * satisfied. If no such partition exists, the multiset charge configuration associated with the requirements may be
+     * rejected.
+     */
+    uint64_t validity_witness_partitioning_max_cluster_size_gss = 6;
+    /**
+     * The complexity is of validity witness partitioning bounded by a factorial in the number of overlapping witnesses.
+     * This parameter thus allows the validity witness partitioning procedure to perform the reduction to overlapping
+     * witnesses for larger cluster sizes that could be runtime-impairing, then limiting specifically the length of the
+     * input to the factorial call.
+     */
+    uint64_t num_overlapping_witnesses_limit_gss = 6;
+    /**
+     * Number of threads to make available to ClusterComplete for the simulation stage.
+     */
+    uint64_t available_threads = std::thread::hardware_concurrency();
+    /**
+     * Report the *Ground State Space* statistics to standard output.
+     */
+    bool report_gss_stats = false;
+};
+
+namespace detail
+{
+
+template <typename Lyt>
+class clustercomplete_impl
+{
+  public:
+    explicit clustercomplete_impl(const Lyt& lyt, const clustercomplete_params<Lyt>& params) noexcept :
+            charge_layout{initialize_charge_layout(lyt, params)},
+            real_placed_defects{charge_layout.get_defects()},
+            mu_bounds_with_error{physical_constants::POP_STABILITY_ERR - params.physical_parameters.mu_minus,
+                                 -physical_constants::POP_STABILITY_ERR - params.physical_parameters.mu_minus,
+                                 physical_constants::POP_STABILITY_ERR - params.physical_parameters.mu_plus(),
+                                 -physical_constants::POP_STABILITY_ERR - params.physical_parameters.mu_plus()},
+            num_threads{params.available_threads}
+    {}
+
+    sidb_simulation_result<Lyt> run(const clustercomplete_params<Lyt>& params) noexcept
+    {
+        res.physical_parameters = charge_layout.get_phys_params();
+        res.algorithm_name      = "ClusterComplete";
+        res.additional_simulation_parameters.emplace("global_potential", params.global_potential);
+        res.additional_simulation_parameters.emplace("validity_witness_partitioning_limit",
+                                                     params.validity_witness_partitioning_max_cluster_size_gss);
+        res.additional_simulation_parameters.emplace("num_overlapping_witnesses_limit",
+                                                     params.num_overlapping_witnesses_limit_gss);
+
+        // run Ground State Space to obtain the complete hierarchical charge space
+        const ground_state_space_stats& gss_stats = fiction::ground_state_space(
```

Comment:
```suggestion
        const ground_state_space_stats& gss_stats = ground_state_space(
```

Reply:
This used to cause a problem without it. Now that some thing changed, it looks like that problem is no more. Thanks!